src/core/lib/security/credentials/credentials.h

```diff
@@ -107,6 +107,9 @@ typedef struct {
       grpc_channel_credentials *c, grpc_call_credentials *call_creds,
       const char *target, const grpc_channel_args *args,
       grpc_channel_security_connector **sc, grpc_channel_args **new_args);
+
+  grpc_channel_credentials *(*duplicate_without_call_credentials)(
```

Comment:
When the channel_credentials is specified from the application code, you don't know whether this will be a composite channel creds with both a channel creds and a call creds or just a *pure* channel creds.
So we had 2 options here:
1. Put this method just on the composite creds with the caller having to make sure (equivalent to a dynamic cast) that the creds is a composite creds.
2. Add this method on the vtable where the caller doesn't need to do a cast.

I chose 2. since that seems much more elegant to me.